ts/components/dialog/debug/components.tsx

```diff
@@ -85,6 +86,63 @@ const CheckVersionButton = ({ channelToCheck }: { channelToCheck: ReleaseChannel
   );
 };
 
+const ForceUpdateButton = () => {
+  const [loading, setLoading] = useState(false);
+  const state = useAsync(async () => {
+    const userEd25519KeyPairBytes = await UserUtils.getUserED25519KeyPairBytes();
+    const userEd25519SecretKey = userEd25519KeyPairBytes?.privKeyBytes;
+    return userEd25519SecretKey;
+  });
+
+  const handleForceUpdate = async () => {
+    window.log.warn(
+      '[updater] [debugMenu] Triggering force update. Current version',
+      window.getVersion()
+    );
+    setLoading(true);
+
+    if (state.loading || state.error) {
+      window.log.error(
+        `[updater] [debugMenu] userEd25519SecretKey loading ${state.loading} error ${state.error}`
+      );
+      setLoading(false);
+      return;
+    }
+
+    if (!state.value) {
+      window.log.error(`[updater] [debugMenu] userEd25519SecretKey not found`);
+      setLoading(false);
+      return;
+    }
+
+    const newVersion = await fetchLatestRelease.fetchReleaseFromFSAndUpdateMain(state.value, true);
```

Comment:
can this throw? If yes we should probably have a try/catch to setLoading(false)

Reply:
Fixed